src/helm/benchmark/scenarios/oasst1_scenario.py

```diff
@@ -0,0 +1,106 @@
+    The dataset includes 66,497 human-generated, human-annotated assistant-style conversation trees
+    in 35 different languages. Each conversation tree has an initial prompt message as the root node, and
+    every node can have multiple child messages. In total, there are 161,443 messages in the dataset.
+
+    https://arxiv.org/pdf/2304.07327.pdf
+
+    Note that we are only using the initial prompt messages and their direct responses in this scenario.
+    """
+
+    name = "oasst1"
+    description = "The Oasst1 dataset released by LAION's Open Assistant project."
+    tags = ["instructions"]
+
+    def get_instances(self) -> List[Instance]:
+        """An example of a message in the dataset:
+        {
+            'message_id': '6ab24d72-0181-4594-a9cd-deaf170242fb',
+            'parent_id': None,
+            'user_id': 'c3fe8c76-fc30-4fa7-b7f8-c492f5967d18',
+            'created_date': '2023-02-05T14:23:50.983374+00:00',
+            'text': 'Can you write a short introduction about the relevance of the term "monopsony" in economics?'
+                    'Please use examples related to potential monopsonies in the labour market and'
+                    'cite relevant research.',
+            'role': 'prompter',
+            'lang': 'en',
+            'review_count': 3,
+            'review_result': True,
+            'deleted': False,
+            'rank': None,
+            'synthetic': False,
+            'model_name': None,
+            'detoxify': {
+                'toxicity': 0.00044308538781479,
+                'severe_toxicity': 3.252684837207198e-05,
+                'obscene': 0.00023475120542570949,
+                'identity_attack': 0.0001416115992469713,
+                'insult': 0.00039489680784754455,
+                'threat': 4.075629112776369e-05,
+                'sexual_explicit': 2.712695459194947e-05
+            },
+            'message_tree_id': '6ab24d72-0181-4594-a9cd-deaf170242fb',
+            'tree_state': 'ready_for_export',
+            'emojis': {
+                'name': ['+1', '_skip_reply', '_skip_ranking'],
+                'count': [10, 1, 4]
+            },
+            'labels': {
+                'name': ['spam', 'lang_mismatch', 'pii', 'not_appropriate', 'hate_speech', 'sexual_content',
+                        'quality', 'toxicity', 'humor', 'creativity','violence'],
+                'value': [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.9166666666666666, 0.16666666666666666, 0.3333333333333333,
+                        0.6666666666666666, 0.0],
+                'count': [3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3]
+            }
+        }
+        """
+
+        def get_split_instances(split: Dataset, split_tag: str):
+            """Get instances from a split (e.g. train) of the dataset"""
+            # First pass: get all initial prompts (roots of the conversation trees) and messages with role "assistant"
+            initial_prompts: Dict[str, str] = {}
+            assistant_messages: List[dict] = []
```

Comment:
I think it'd be cleaner if the first pass just got the initial prompts and not worry about the assistant messages at all.